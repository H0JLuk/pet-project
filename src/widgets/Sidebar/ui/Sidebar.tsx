import { useState } from 'react';
import { useTranslation } from 'react-i18next';

import { classNames } from '@/shared/lib/classNames';
import { LangSwitcher } from '@/shared/ui/LangSwitcher';
import { ThemeSwitcher } from '@/shared/ui/ThemeSwitcher';

import cls from './Sidebar.module.scss';
import { Button } from '@/shared/ui/Button';

export type SidebarProps = {
  className?: string;
  testId?: string;
};

export const Sidebar = ({ className, testId = 'sidebar' }: SidebarProps) => {
  const [collapsed, setCollapsed] = useState(false);

  const { t } = useTranslation();

  const onToggle = () => setCollapsed(prev => !prev);

  return (
    <div className={classNames(cls.sidebar, { [cls.collapsed]: collapsed }, [className])} data-testid={testId}>
      <Button onClick={onToggle} data-testid='sidebar-toggle-btn'>
        {t('Свернуть')}
      </Button>
      <div className={cls.switchers}>
        <ThemeSwitcher />
        <LangSwitcher />
      </div>
    </div>
  );
};
