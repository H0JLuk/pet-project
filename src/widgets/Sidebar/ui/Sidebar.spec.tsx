import { renderWithTranslation } from '@/shared/lib/tests/renderWithTranslation/renderWithTranslation';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { withTranslation } from 'react-i18next';

import { Sidebar, SidebarProps } from './Sidebar';

const baseProps = { testId: 'sidebar-test-id' } satisfies SidebarProps;

const renderComponent = (props: SidebarProps = {}) => {
  const mergedProps: SidebarProps = { ...baseProps, ...props };

  return renderWithTranslation(<Sidebar {...mergedProps} />);
};

describe('<Sidebar />', () => {
  test('should be match snapshot', () => {
    renderComponent();
    expect(screen.getByTestId(baseProps.testId)).toBeInTheDocument();
    // expect(screen.debug()).toMatchSnapshot();
  });

  test('should be collapse', async () => {
    renderComponent();
    const toggleBtn = screen.getByTestId('sidebar-toggle-btn');

    expect(screen.getByTestId(baseProps.testId)).not.toHaveClass('collapsed');

    await userEvent.click(toggleBtn);
    expect(screen.getByTestId(baseProps.testId)).toHaveClass('collapsed');

    await userEvent.click(toggleBtn);
    expect(screen.getByTestId(baseProps.testId)).not.toHaveClass('collapsed');
  });
});
