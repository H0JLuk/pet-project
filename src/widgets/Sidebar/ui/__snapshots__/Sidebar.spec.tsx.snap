// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> should be match snapshot 1`] = `undefined`;

exports[`<Sidebar /> should be match snapshot 2`] = `undefined`;
